heft-plugins/heft-webpack5-plugin/src/WebpackPlugin.ts

```diff
@@ -1,170 +1,200 @@
 // Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license.
 // See LICENSE in the project root for license information.
 
-import * as nodePath from 'path';
-import type {
-  Compiler as WebpackCompiler,
-  MultiCompiler as WebpackMultiCompiler,
-  Stats as WebpackStats,
-  MultiStats as WebpackMultiStats,
-  StatsCompilation as WebpackStatsCompilation,
-  StatsError as WebpackStatsError
-} from 'webpack';
+import { AsyncParallelHook, AsyncSeriesWaterfallHook } from 'tapable';
+import type * as TWebpack from 'webpack';
 import type TWebpackDevServer from 'webpack-dev-server';
-import { LegacyAdapters, Path, Import, IPackageJson, PackageJsonLookup } from '@rushstack/node-core-library';
+import {
+  FileError,
+  Import,
+  IPackageJson,
+  LegacyAdapters,
+  PackageJsonLookup
+} from '@rushstack/node-core-library';
 import type {
   HeftConfiguration,
-  HeftSession,
-  IBuildStageContext,
-  IBuildStageProperties,
-  IBundleSubstage,
-  IHeftPlugin,
-  ScopedLogger
+  IHeftTaskSession,
+  IHeftTaskCleanHookOptions,
+  IHeftTaskPlugin,
+  IHeftTaskRunHookOptions,
+  IScopedLogger
 } from '@rushstack/heft';
+
 import type {
   IWebpackConfiguration,
-  IWebpackBundleSubstageProperties,
-  IWebpackBuildStageProperties
+  IWebpackConfigurationWithDevServer,
+  IWebpackPluginAccessor,
+  IWebpackVersions
 } from './shared';
 import { WebpackConfigurationLoader } from './WebpackConfigurationLoader';
 
-const webpack: typeof import('webpack') = Import.lazy('webpack', require);
-
-const PLUGIN_NAME: string = 'WebpackPlugin';
+/**
+ * @public
+ */
+export const PLUGIN_NAME: string = 'WebpackPlugin';
 const WEBPACK_DEV_SERVER_PACKAGE_NAME: string = 'webpack-dev-server';
 const WEBPACK_DEV_SERVER_ENV_VAR_NAME: string = 'WEBPACK_DEV_SERVER';
 
-interface IWebpackVersions {
-  webpackVersion: string;
-  webpackDevServerVersion: string;
-}
-
 /**
  * @internal
  */
-export class WebpackPlugin implements IHeftPlugin {
-  public readonly pluginName: string = PLUGIN_NAME;
+export default class WebpackPlugin implements IHeftTaskPlugin {
```

Comment:
I didn't read through this in detail because I assume it's pretty much the same as the webpack 4 version. Anything in particular that's different?

Reply:
Nope, they're pretty much identical.